test/interop/hyperlane_test.go

```diff
@@ -0,0 +1,269 @@
+package interop
+
+import (
+	"fmt"
+	"testing"
+
+	"cosmossdk.io/math"
+	"github.com/bcp-innovations/hyperlane-cosmos/util"
+	ismtypes "github.com/bcp-innovations/hyperlane-cosmos/x/core/01_interchain_security/types"
+	hooktypes "github.com/bcp-innovations/hyperlane-cosmos/x/core/02_post_dispatch/types"
+	coretypes "github.com/bcp-innovations/hyperlane-cosmos/x/core/types"
+	warptypes "github.com/bcp-innovations/hyperlane-cosmos/x/warp/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/gogoproto/proto"
+	ibctesting "github.com/cosmos/ibc-go/v8/testing"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/celestiaorg/celestia-app/v4/app"
+	"github.com/celestiaorg/celestia-app/v4/app/params"
+	minttypes "github.com/celestiaorg/celestia-app/v4/x/mint/types"
+)
+
+type HyperlaneTestSuite struct {
+	suite.Suite
+
+	celestia *ibctesting.TestChain
+	simapp   *ibctesting.TestChain
+}
+
+func TestHyperlaneTestSuite(t *testing.T) {
+	suite.Run(t, new(HyperlaneTestSuite))
+}
+
+// TODO: refactor the test setup for hyperlane, pfm and tokenfilter suites
```

Comment:
I would like to do a follow up PR to address this so we can reduce the diff in this PR.

With renaming this package from pfm -> interop, the intension is to:
- Reuse the simapp defined in this package
- Create a `PacketForwardMiddlewareTestSuite` (similarly to how it is done for hyperlane here)
- Create a `TokenFilterTestSuite` (similarly ^^)

This will keep interop integration testing together where they can reuse the simapp for a counterparty to the celestia app.

Reply:
+100 this sounds like a great idea. Thanks for suggesting and reusing the simapp in this package for this test. Seems easier to maintain the interop tests given the code reuse.